cms/djangoapps/contentstore/views/item.py

```diff
@@ -61,6 +63,8 @@
 
 CREATE_IF_NOT_FOUND = ['course_info']
 
+COMPONENT_TYPES = sorted(set(name for name, class_ in XBlock.load_classes()) - set(DIRECT_ONLY_CATEGORIES))
```

Comment:
I don't recommend making this a constant. It isn't expensive to compute it on demand, and new XBlocks can in theory get loaded after this line gets evaluated.